src/background_listener.rs

```diff
@@ -36,18 +35,25 @@ pub struct WindowVisibility(bool);
 
 /// The `system_init_active_background_listener` system initializes the visibilitychange listener which runs the `Main` schedule once when hidden
 fn system_init_active_background_listener(world: &mut World) {
-    let world_ptr = Arc::new(RwLock::new(world as *mut World));
-
-    let closure = Closure::<dyn FnMut()>::new(move || {
-        let world = unsafe { world_ptr.write().unwrap().as_mut().unwrap() };
-        let window = window().expect("Unable to access the window");
-        let document = window
-            .document()
-            .expect("Unable to access the document, is the app running in headless mode?");
-        let is_hidden = document.hidden();
-        world.resource_mut::<WindowVisibility>().0 = !is_hidden;
-        if is_hidden {
-            world.run_schedule(Main);
+    let world_ptr = Rc::new(world as *mut World);
+    let closure = Closure::<dyn FnMut()>::new({
+        let world = world_ptr.clone();
+        move || {
+            let window = window().expect("Unable to access the window");
+            let document = window
+                .document()
+                .expect("Unable to access the document, is the app running in headless mode?");
+            let is_hidden = document.hidden();
+
+            unsafe {
+                let Some(world) = world.as_mut() else {
```

Comment:
no need really, this should never fail unless the world is dropped (which doesnt happen over the entire lifecycle of the app)

Reply:
That's right, I think I'd rather take the approach to remove the event listener/callback if the world is dropped.

Reply:
actually i guess this could happen if the apps lifetime doesnt match the lifetime of the listener, but this is kind of unrealistic so im prob not gonna address that unless absolutely needed. lets keep this for now.

Reply:
> That's right, I think I'd rather take the approach to remove the event listener/callback if the world is dropped.

sounds like a pain to implement tbh since we would need to keep a ref of the closure somewhere in the world, so that we can remove the eventlistener when its dropped. However thats not possible i think because we need to .forget() it.